cmd/device-worker/main.go

```diff
@@ -97,11 +98,21 @@ func main() {
 	}
 	configManager := configuration2.NewConfigurationManager(dataDir)
 
+	// --- Client metrics configuration ---
+	metricsStore, err := metrics.NewTSDB(dataDir)
+	configManager.RegisterObserver(metricsStore)
+	// Metrics Daemon
+	metricsDaemon := metrics.NewMetricsDaemon(metricsStore)
+	workloadMetricWatcher := metrics.NewWorkloadMetrics(metricsDaemon)
+	configManager.RegisterObserver(workloadMetricWatcher)
+	go metricsDaemon.Start()
```

Comment:
Is there a need for clean stop? Should we handle termination signals?

Reply:
It's part of the Start function. If it's already started will  stop it!